pennylane/qnn.py

```diff
@@ -0,0 +1,205 @@
+
+        weight_shapes = {"weights": (3, n_qubits, 3)}
+        weight_specs = {"weights": {"initializer": "random_uniform"}}
+
+        qnode = qml.qnn.KerasLayer(circuit, weight_shapes, 2, weight_specs)
+
+    The resulting ``qnode`` can be treated as a Keras layer and can be combined with other layers
+    using the `Sequential <https://www.tensorflow.org/api_docs/python/tf/keras/Sequential>`__ or
+    `Model <https://www.tensorflow.org/api_docs/python/tf/keras/Model>`__ APIs.
+
+    Args:
+        qnode (qml.QNode): the PennyLane QNode to be converted into a Keras layer
+        weight_shapes (dict[str, tuple]): a dictionary mapping from all weights used in the QNode to
+            their corresponding sizes
+        output_dim (int): the dimension of data output from the QNode
+        weight_specs (dict[str, dict]): an optional dictionary for users to provide additional
+            specifications for weights used in the QNode, such as the method of parameter
+            initialization. This specification is provided as a dictionary with keys given by the
+            arguments of the `add_weight()
+            <https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#add_weight>`__
+            (e.g., ``initializer``) method and values being the corresponding specification.
+        **kwargs: additional keyword arguments passed to the `Layer
+            <https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer>`__ base class
+    """
+
+    def __init__(
+        self,
+        qnode: QNode,
+        weight_shapes: dict,
+        output_dim: int,
+        weight_specs: Optional[dict] = None,
+        **kwargs
+    ):
+        if not CORRECT_TF_VERSION:
+            raise ImportError("KerasLayer requires TensorFlow version 2 and above")
+
+        self.sig = qnode.func.sig
+
+        if INPUT_ARG not in self.sig:
+            raise TypeError(
+                "QNode must include an argument with name {} for inputting data".format(INPUT_ARG)
+            )
+        if INPUT_ARG in set(weight_shapes.keys()):
+            raise ValueError(
+                "{} argument should not have its dimension specified in "
+                "weight_shapes".format(INPUT_ARG)
+            )
+        if set(weight_shapes.keys()) | {INPUT_ARG} != set(self.sig.keys()):
+            raise ValueError("Must specify a shape for every non-input parameter in the QNode")
+        if qnode.func.var_pos:
+            raise TypeError("Cannot have a variable number of positional arguments")
+        if qnode.func.var_keyword:
+            raise TypeError("Cannot have a variable number of keyword arguments")
+
+        self.qnode = to_tf(qnode, dtype=tf.keras.backend.floatx())
+        self.weight_shapes = {
+            weight: (tuple(size) if isinstance(size, Iterable) else (size,))
+            for weight, size in weight_shapes.items()
+        }
+        self.output_dim = output_dim[0] if isinstance(output_dim, Iterable) else output_dim
```

Comment:
How come the first element of the output dimension is taken if output_dim is an iterable? Shouldn't output_dim always be an int?

Reply:
It might be worth commenting this in there so that future devs understand why this deviates from the docstring/type hint

Reply:
Edit: can't see from this view whether `output_dim` is using type hints or not, but if it is `output_dim: int`, I would suggest removing it since here we are taking good advantage of Python's duck typing.

Reply:
Ok, added the comment and removed the type hint :+1: thanks!